platform-services/src/std_platform_services.rs

```diff
@@ -15,14 +17,48 @@
 use super::result;
 
 use getrandom;
+use nix::{
+    time,
+    sys::time::TimeValLike,
+};
 
 /// Fills a buffer, `buffer`, with random bytes sampled from the random number
 /// source provided by the host operating system, as provided by `getrandom`.
 ///
 /// This is for use with "freestanding-execution-engine".
-pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result {
+pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result<()> {
     if let Ok(_) = getrandom::getrandom(buffer) {
-        return result::Result::Success;
+        return result::Result::Success(());
     }
     result::Result::UnknownError
 }
+
+/// Returns the clock resolution in nanoseconds.
+///
+/// This is for use with "freestanding-execution-engine".
+pub fn platform_getclockres(clock_id: u8) -> result::Result<u64> {
+    let clock_id = time::ClockId::from_raw(clock_id.into());
+    let timespec = match time::clock_getres(clock_id) {
+        Ok(t) => t,
+        Err(_) => return result::Result::Unavailable,
+    };
+
+    // Catch overflow
+    if timespec.tv_sec() == 0 {
```

Comment:
Don't you expect `tv_sec` to be zero in the result from `clock_getres`?

Reply:
Here's the source for `num_nanoseconds`: https://github.com/nix-rust/nix/blob/master/src/sys/time.rs#L173

Doesn't that already handle `tv_sec` for you?

Reply:
Thanks for the spotting.
Actually I don't think there's a point in catching overflows here, as timespec.tv_nsec is an i64